ansible_collections/arista/avd/roles/eos_designs/python_modules/network_services/router_bgp.py

```diff
@@ -406,20 +413,37 @@ def _router_bgp_vlan_aware_bundles(self) -> list | None:
                     )
 
             # L2 Vlans per Tenant
-            # If multiple L2 Vlans share the same name, they will be part of the same bundle
-            for bundle_name, l2vlans in groupby(tenant["l2vlans"], "name"):
+            # If multiple L2 Vlans share the same evpn_vlan_bundle name, they will be part of the same vlan-aware-bundle else they use the vlan name as bundle
+            sorted_vlan_list = sorted(tenant["l2vlans"], key=self._get_vlan_aware_bundle_name)
+            bundle_groups = itertools_groupby(sorted_vlan_list, self._get_vlan_aware_bundle_name)
+            for bundle_name, l2vlans in bundle_groups:
                 l2vlans = list(l2vlans)
                 if (bundle := self._router_bgp_vlans_vlan(l2vlans[0], tenant, vrf={})) is not None:
                     # We are reusing the regular bgp vlan function so need to add vlan info
                     bundle["vlan"] = list_compress([int(l2vlan["id"]) for l2vlan in l2vlans])
                     bundle = {"name": bundle_name, **bundle}
-                    append_if_not_duplicate(
-                        list_of_dicts=bundles,
-                        primary_key="name",
-                        new_dict=bundle,
-                        context="BGP VLAN-Aware Bundles defined under network services",
-                        context_keys=["name"],
-                    )
+
+                    if "evpn_vlan_bundle" in l2vlans[0]:
+                        # If "evpn_vlan_bundle" in l2vlan bundle group, check if the referred name exists in the global evpn_vlan_bundles
+                        if (evpn_vlan_bundle := get_item(self._hostvars["evpn_vlan_bundles"], "name", l2vlans[0]["evpn_vlan_bundle"])) is not None:
+                            # override the settings from l2vlan
+                            bundle["rd"] = self.get_vlan_rd(evpn_vlan_bundle, tenant)
+                            bundle["route_targets"] = {"both": [self.get_vlan_rt(evpn_vlan_bundle, tenant)]}
+                            if (eos_cli := get(evpn_vlan_bundle, "bgp.raw_eos_cli")) is not None:
+                                bundle["eos_cli"] = eos_cli
+                        else:
+                            raise AristaAvdMissingVariableError(
+                                "The 'evpn_vlan_bundle' of the l2vlans must be defined in the global 'evpn_vlan_bundles' setting. First occurence seen for"
+                                f" l2vlan {l2vlans[0]['id']} in Tenant '{l2vlans[0]['tenant']}' and evpn_vlan_bundle '{l2vlans[0]['evpn_vlan_bundle']}'."
+                            )
+
+                append_if_not_duplicate(
```

Comment:
I think you deindented this line by mistake, since now it will also run in case the `if` in line 421 fails.

Reply:
changed